### ISSUE TITLE
PANIC: NodeRef controller

### ISSUE THREAD

**Issue by Sn0rt** [kubernetes-sigs/cluster-api-provider-aws#929]
/kind bug

```
E0724 06:43:22.244531       1 controller.go:217] kubebuilder/controller "msg"="Reconciler error" "error"="Get https://test1-apiserver-561763734.us-east-2.elb.amazonaws.com:6443/api/v1/nodes: EOF"  "controller"="noderef-controller" "request"={"Namespace":"default","Name":"test1-controlplane-0"}
I0724 06:43:46.620426       1 noderef_controller.go:84] Reconcile request for Machine "test1-node-gjxww" in namespace "default"
W0724 06:43:46.620523       1 noderef_controller.go:129] Machine "test1-node-gjxww" in namespace "default" doesn't have a valid ProviderID, retrying later
I0724 06:44:16.620741       1 noderef_controller.go:84] Reconcile request for Machine "test1-node-gjxww" in namespace "default"
W0724 06:44:16.620818       1 noderef_controller.go:129] Machine "test1-node-gjxww" in namespace "default" doesn't have a valid ProviderID, retrying later
I0724 06:44:44.164698       1 noderef_controller.go:84] Reconcile request for Machine "test1-controlplane-0" in namespace "default"
I0724 06:44:44.255733       1 noderef_controller.go:140] Set Machine's ("test1-controlplane-0" in namespace "default") NodeRef to "ip-10-0-0-34.us-east-2.compute.internal"
I0724 06:44:44.258055       1 noderef_controller.go:84] Reconcile request for Machine "test1-controlplane-0" in namespace "default"
I0724 06:44:46.621074       1 noderef_controller.go:84] Reconcile request for Machine "test1-node-gjxww" in namespace "default"
W0724 06:44:46.621214       1 noderef_controller.go:129] Machine "test1-node-gjxww" in namespace "default" doesn't have a valid ProviderID, retrying later
I0724 06:45:16.621447       1 noderef_controller.go:84] Reconcile request for Machine "test1-node-gjxww" in namespace "default"
W0724 06:45:16.621749       1 noderef_controller.go:129] Machine "test1-node-gjxww" in namespace "default" doesn't have a valid ProviderID, retrying later


I0724 06:45:46.621992       1 noderef_controller.go:84] Reconcile request for Machine "test1-node-gjxww" in namespace "default"
W0724 06:45:46.622070       1 noderef_controller.go:129] Machine "test1-node-gjxww" in namespace "default" doesn't have a valid ProviderID, retrying later


I0724 06:45:49.486991       1 noderef_controller.go:84] Reconcile request for Machine "test1-node-gjxww" in namespace "default"
W0724 06:45:49.487320       1 noderef_controller.go:129] Machine "test1-node-gjxww" in namespace "default" doesn't have a valid ProviderID, retrying later
I0724 06:45:49.492638       1 noderef_controller.go:84] Reconcile request for Machine "test1-node-gjxww" in namespace "default"
W0724 06:45:49.492674       1 noderef_controller.go:129] Machine "test1-node-gjxww" in namespace "default" doesn't have a valid ProviderID, retrying later
I0724 06:45:49.632427       1 noderef_controller.go:84] Reconcile request for Machine "test1-node-gjxww" in namespace "default"
W0724 06:45:49.636668       1 noderef_controller.go:165] Cannot find a matching Node for Machine "test1-node-gjxww" in namespace "default", retrying later
E0724 06:45:49.636787       1 runtime.go:69] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/go/src/sigs.k8s.io/cluster-api/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
/go/src/sigs.k8s.io/cluster-api/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/sigs.k8s.io/cluster-api/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/panic.go:522
/usr/local/go/src/runtime/panic.go:82
/usr/local/go/src/runtime/signal_unix.go:390
/go/src/sigs.k8s.io/cluster-api/pkg/controller/noderef/noderef_controller.go:140
/go/src/sigs.k8s.io/cluster-api/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215
/go/src/sigs.k8s.io/cluster-api/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158
/go/src/sigs.k8s.io/cluster-api/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/sigs.k8s.io/cluster-api/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/sigs.k8s.io/cluster-api/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:1337
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x10ec52f]

goroutine 199 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/sigs.k8s.io/cluster-api/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x105
panic(0x12084a0, 0x1fbe250)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
sigs.k8s.io/cluster-api/pkg/controller/noderef.(*ReconcileNodeRef).Reconcile(0xc00013e480, 0xc0007fecc9, 0x7, 0xc0007fec90, 0x10, 0x1fd20e0, 0x6fc23ac00, 0x0, 0x0)
	/go/src/sigs.k8s.io/cluster-api/pkg/controller/noderef/noderef_controller.go:140 +0x8cf
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0000b6e60, 0x0)
	/go/src/sigs.k8s.io/cluster-api/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215 +0x1cc
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1()
	/go/src/sigs.k8s.io/cluster-api/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158 +0x36
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0005e0720)
	/go/src/sigs.k8s.io/cluster-api/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0005e0720, 0x3b9aca00, 0x0, 0xc000000001, 0xc0002e5ec0)
	/go/src/sigs.k8s.io/cluster-api/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc0005e0720, 0x3b9aca00, 0xc0002e5ec0)
	/go/src/sigs.k8s.io/cluster-api/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	/go/src/sigs.k8s.io/cluster-api/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:157 +0x311
rpc error: code = Unknown desc = Error: No such container: bb350ea66b6955ebc1a4bb04534c7f3b9a4413d48bc3e3b22165268535011198%
```
**Environment:**

- Cluster-api-provider-aws version: 
`us.gcr.io/k8s-artifacts-prod/cluster-api/cluster-api-controller:v0.1.6`

- Kubernetes version: (use `kubectl version`): 
```
guohao@buffer ~/workspace/aws $ kubectl version
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.1", GitCommit:"eec55b9ba98609a46fee712359c7b5b365bdd920", GitTreeState:"clean", BuildDate:"2018-12-13T10:39:04Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.8", GitCommit:"a89f8c11a5f4f132503edbc4918c98518fd504e3", GitTreeState:"clean", BuildDate:"2019-04-23T04:41:47Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
```
- OS (e.g. from `/etc/os-release`):

**Comment by Sn0rt** [comment]
https://github.com/kubernetes-sigs/cluster-api/issues/1104

**Comment by ncdc** [comment]
This will be fixed in the next v0.3.x release, which has cluster-api v0.1.8.

/assign @detiber

**Comment by detiber** [comment]
This should be resolved now that [v0.3.6](https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases/tag/v0.3.6) has been released
